### PR TITLE
backlog: ticket templates should recommend to update rclone

### PR DESCRIPTION
**Pull request by ivandeex**
#### What is the purpose of this change?

This PR partially implements #5074 
It adds useful recommendations from the forum ticket templates to the github issue templates
and recommends users to run `rclone selfupdate`.

It blocks on #5080 but should be merged separately
#5080 can be merged after successful review but will be released with rclone 1.55
This PR will affect rclone github templates immediately after merge on master.

Let's review this now but merge together with 1.55 release

#### Was the change discussed in an issue or in the forum before?

#5074
https://forum.rclone.org/t/previously-working-rclone-backup-now-giving-error-unknown-command-rcat-for-rclone/22507/14

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by ivandeex** [comment]
code quality test break until someone reviews #5093

**Comment by ivandeex** [comment]
@darthShadow 

Here is your comments from https://github.com/rclone/rclone/issues/5074#issuecomment-787424433

> We should probably also update the Bug template to be similar to the "Help & Support" template on the forum.

> Ideally, only issues that have been confirmed on the forum should be created here so perhaps a bot that enforces the presence of a valid forum URL in the issue body, without which the issue is closed automatically?

> That may be too drastic a step, however, so perhaps the easiest/simplest thing would be to simply copy the forum template pasted above and add an additional line to post on the forum first?

I did not yet make it **first**. Should I?

@darthShadow

**Comment by ivandeex** [comment]
@ncw @darthShadow Could you please review this?

**Comment by darthShadow** [comment]
Sorry for the delay. Been a hectic week at work.

> I did not yet make it the first. Should I?

Yeah, I still believe that it should be the first after the following lines:

```
**STOP and READ**
**YOUR POST WILL BE REMOVED IF IT IS LOW QUALITY**:
Please show the effort you've put in to solving the problem and please be specific.
People are volunteering their time to help you! Low effort issues are not likely to get good answers!
```

**Comment by ivandeex** [comment]
> > I did not yet make it the first. Should I?
> Yeah, I still believe that it should be the first after the following lines:

After reviewing full bug template _with the current patch applied_ I can see that it will be _at the top_, above those lines:
- bug request: https://raw.githubusercontent.com/rclone/rclone/457dc17b5bccad48240ed8345067f8394b17e337/.github/ISSUE_TEMPLATE/Bug.md
- feature request:  https://raw.githubusercontent.com/rclone/rclone/457dc17b5bccad48240ed8345067f8394b17e337/.github/ISSUE_TEMPLATE/Feature.md

So I think you requirement will be satisfied. :smile_cat:

**Comment by ivandeex** [comment]
BTW #5080 has been merged yesterday. This PR now waits only for the 1.55 release.

**Comment by darthShadow** [comment]
> So I think you requirement will be satisfied.

I meant that these following lines should be above the "Welcome" statement.

```
**STOP and READ**
**YOUR ISSUE WILL BE REMOVED IF IT IS LOW QUALITY**:
Please show the effort you've put in to solving the problem and please be specific.
People are volunteering their time to help you! Low effort posts are not likely to get good answers!
```

Something like this: https://gist.githubusercontent.com/darthShadow/fc825e528e6625253793939b4ccb1ac3/raw/9d795db6c9c39aca0072cfe008164f45401b7cbe/Bug.md

**Comment by ivandeex** [comment]
Done.

Now the bug template looks like this https://raw.githubusercontent.com/rclone/rclone/a7a5264468a8005c1f0f67dc83c3711197d7bba6/.github/ISSUE_TEMPLATE/Bug.md

I intentionally restored welcoming line to balance up the overall tone.
Also replaced the second "to help you!" with "to help!". This removes duplication and makes it sound milder IMO.

**Comment by ivandeex** [comment]
@darthShadow 

Added a link to the forum in the bug and feature templates.
Current results:
- your _bug_ template proposal: https://gist.githubusercontent.com/darthShadow/fc825e528e6625253793939b4ccb1ac3/raw/9d795db6c9c39aca0072cfe008164f45401b7cbe/Bug.md
- resulting _bug_ template from this PR: https://raw.githubusercontent.com/rclone/rclone/14ca104379a7158186caa009d5d34c2ef2c621f2/.github/ISSUE_TEMPLATE/Bug.md
- resulting _feature_ template from this PR: https://raw.githubusercontent.com/rclone/rclone/14ca104379a7158186caa009d5d34c2ef2c621f2/.github/ISSUE_TEMPLATE/Feature.md

What do you think?

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ncw) [inline .github/ISSUE_TEMPLATE/Bug.md:26]: Do you think we should add something like

If your rclone reports `unknown command "self-update"` then you'll need to update manually following the [install instructions](https://rclone.org/install/)

?

PS I think the install instructions should probably mention self-update too :-)

**Review by ncw — COMMENTED** [review]
I think this looks good :-)

One comment inline

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:26]: > I think the install instructions should probably mention self-update too

Please review https://github.com/rclone/rclone/pull/5122 :pray:

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Bug.md:21]: This should probably be moved before the above section about going to the forum.

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Bug.md:70]: This should also probably be the first question or even the only question since the forum post already has the other requested information.

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Feature.md:22]: Good idea to also specify the template to be used in the forum i.e. `Feature` or `Dev Discussion` depending on the type of request and the level of involvement the user wants to have.

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Bug.md:19]: We also have the `Suspected Bug` Template for suspected bugs instead of `Help & Support` which is targeted for general usage questions.

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:70]: So we visually increase forum importance. Agreed. (2)

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:21]: So we visually increase forum importance. Agreed.

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Feature.md:22]: Can we leave this to the forum template?

Comment (darthShadow) [reply .github/ISSUE_TEMPLATE/Feature.md:22]: Fine either way but if we are not mentioning the template here then we should probably remove the mention of the `Help & Support Template` from the Bug template too.

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (darthShadow) [reply .github/ISSUE_TEMPLATE/Bug.md:21]: This was meant more to make sure they try the latest version before even posting on the forum.

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:19]: I am in favor of removing the `Help & Support Template` detail from the _github_ template. Let forum admins have freedom to improve their templates without need to chase their changes here.

Comment (darthShadow) [reply .github/ISSUE_TEMPLATE/Bug.md:19]: Sounds good.

**Comment by ivandeex** [comment]
New templates after 2nd review by @darthShadow: [Bug.md](https://raw.githubusercontent.com/rclone/rclone/3f701754d2572c01a17a89192d5af897a9eb1d7d/.github/ISSUE_TEMPLATE/Bug.md), [Feature.md](https://raw.githubusercontent.com/rclone/rclone/3f701754d2572c01a17a89192d5af897a9eb1d7d/.github/ISSUE_TEMPLATE/Feature.md)

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Bug.md:15]: At a second glance, should we leave it to the user for deciding to switch to either the latest stable or beta version to try?

And also, we should probably indicate that the operation cannot be reversed (at least not without noting down the previous version, which sounds like a good idea to print while doing the update).

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Bug.md:32]: Should probably be on a separate line to emphasize it.

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Feature.md:22]: 2 & 1 should probably be inter-changed.

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (darthShadow) [inline .github/ISSUE_TEMPLATE/Feature.md:14]: Should we ask to update to `beta` instead? The same conditions apply regarding the choice and revert that was mentioned for the bug template.

**Review by darthShadow — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Feature.md:22]: Don't we give priority to the forum now?

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:32]: ok

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (darthShadow) [reply .github/ISSUE_TEMPLATE/Feature.md:22]: Yes, we do, but if there is an issue/FR already present, then creating a duplicate post about it won't help.

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:15]: > At a second glance, should we leave it to the user for deciding to switch to either the latest stable or beta version to try?
The text does leave it to the user. Should I replace "latest release" with "latest `stable` release"?

> And also, we should probably indicate that the operation cannot be reversed.

Add `You might want to backup your current rclone executable before the update` before `If your rclone reports `unknown command "self-update"` then you'll need to update manually following the install instructions at https://rclone.org/install`
??

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Feature.md:14]: Beta features can be buggy and are not guaranteed to settle before next release and the news may not be propagated across the team.
Moreover, beta features frequently exist on dedicated git branches rather than on master. A support person will have to recommend a concrete beta anyway.
This is in contrast to the bug fixing template, where *any* fix is better than *no* fix. I thought about removing beta from there too, but I was slightly biased to keep it.

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (darthShadow) [reply .github/ISSUE_TEMPLATE/Bug.md:15]: > The text does leave it to the user

I meant should we leave it or not? Some issues/features may only be fixed/added in the latest beta 
with a point release pending.

> You might want to backup your current rclone executable before the update

What about deb/rpm packages? Ideally, this should be done by self-update where it prints the version before going for the update so that a revert is as easy as specifying that version with the self-update command.

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (darthShadow) [reply .github/ISSUE_TEMPLATE/Feature.md:14]: Alright, that makes sense.

File: .github/ISSUE_TEMPLATE/Feature.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Feature.md:22]: ok. will swap back

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:15]: @darthShadow 

> I meant should we leave it or not? Some issues/features may only be fixed/added in the latest beta
> with a point release pending.

Let's look at the context around this template.

There are two groups of users submitting bug reports:
1. Home/office rclone users ("normal" users)
2. Developers who understand rclone internals, maybe from the rclone team

Professionals will most probably know for sure what they want. The template is targeted at an average user.

The user will see this template when he/she submits a new bug request on github.
They can come to this point  through three ways:
1. They found a bug in rclone all alone
2. They come from the forum after a number of discussions with rclone support
3. They come from another github issue after discussions with rclone developers or other users

Users coming through ways 2 and 3 already have some background on the issue they report.
They know when and where the problem manifests itself.
Probably they had been given a prior advice on what version to try and they will follow it rather than a generic boilerplate.
We are targeting the way no.1 here.

We have the following options:
1. Leave user a _choice_ between latest stable and beta - (as it is in the current template)
2. Keep only _latest stable_ - (as it generally was before this pull request, I will elaborate below)
3. Keep only _latest beta_ - (this is what you want, let's elaborate below)
4. Advice on a _concrete beta_ version - (inappropriate for a generic template)
5. There will never be a "concrete stable option", there will be only "last stable"

No.5 is meaningless.
No.4 is for advanced users coming from forum or from another bug thread on github.

@ncw **Please help! Choosing between 1,2,3 needs more discussion. I can't decide without you!**

> What about deb/rpm packages?

I don't want to make a generic bug template too complex or convert it into a _detailed_ manual page for the selfupdate command. We already have a link from the template to the quick start in https://rclone.org/install which hides install details from the template. Currently even that page bypasses deb/rpm details and links further to the self-update html page for those.

If you want to bring deb/rpm details two levels up right into the bug template, there should be a reason for that.
Currently the deb/rpm package contains a single binary, a MIT license (it never changes) and a manual page. The selfupdate command without the deb/rpm flag will still change the binary _so the user can reproduce their bug_. It will probably leave the local manual out of sync. The self-update command should warn about it. The self-update help page should warn about it. But it's not a duty of the bug template. Its duty is to tell user how to _reproduce and report a bug_.

My goal for this command was to make things simple for end user.
The command hides gory details from user - they don't need to open download.rclone.org in the browser, save archive in a temporary place, unpack, apply permissions. You just type a single command and hit enter.
The command simplifies replies of support - instead of "get the https://xxx.rclone.org/long_url" they can write "please run rclone selfupdate --beta --version xxx".

By adding more details here, by piling up features we lose the very goal of simplicity.
Compare with "restic selfupdate" command which has only one option: "--output PATH" for several releases.
The "rclone selfupdate" has 5 options from the very start. I think it's a wrong direction.

> Ideally, this (backup) should be done by self-update where it prints the version before going for the update so that a revert is as easy as specifying that version with the self-update command.

So the `rclone self-update` will either print `Successfully updated from OLDVER to NEWVER`
(or e.g. `Successfully updated to NEWVER. Use "rclone selfupdate [--beta] --version OLDVER" to restore`).
It's a _good point_ deserving a ticket on its own.  Let's not mix it in this thread or we lose both.

cc @ncw

Comment (darthShadow) [reply .github/ISSUE_TEMPLATE/Bug.md:15]: Alright, then we can probably skip the update command altogether and redirect them to a page on the website with the usage details of self-update. That page will/can contain all the details of how to upgrade and how to revert between stable/beta versions (depending on the choice between 1,2 & 3). This can probably be something as simple as (once we decide on either stable or beta or letting the user decide):

```
If you think you might have found a bug, try to replicate it with the latest beta (or stable). The upgrade instructions are available at https://rclone.org/commands/rclone_selfupdate/#beta (or stable)
```

This will simplify the template and have only a single point of reference everywhere rather than duplicating it across templates.

**Comment by ncw** [comment]
I'm happy for you @ivandeex to merge this when you and @darthShadow are happy :-)

File: .github/ISSUE_TEMPLATE/Bug.md
Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:15]: You are right. :-)
The target page can be seen here until the next release (of course, I will commit without the `tip` part) :
https://tip.rclone.org/commands/rclone_selfupdate/

I mentions deb/rpm packages but lacks details about beta updates in the synopsis section.
It does not explain how to revert either. I will improve its text shortly!
Unfortunately this backlog PR has a special lifecycle, so I will have to do that in a separate PR and ask you to review!

Comment (ivandeex) [reply .github/ISSUE_TEMPLATE/Bug.md:15]: The synopsis part in https://tip.rclone.org/commands/rclone_selfupdate/
should also contain the following clause (I will remove from the Feature template):
```
This command was not available before rclone version 1.55, if your rclone reports
`unknown command "self-update"` then you'll need to update manually
following the install instructions at https://rclone.org/install/
```
Thus we gather all update details in a single point.

**Comment by ivandeex** [comment]
Summary after 3rd review by @darthShadow:
- new Bug template: [Bug.md](https://raw.githubusercontent.com/rclone/rclone/d93eededea45d1073114eb5d68697aeba2f5eafb/.github/ISSUE_TEMPLATE/Bug.md)
- new Feature template [Feature.md](https://raw.githubusercontent.com/rclone/rclone/d93eededea45d1073114eb5d68697aeba2f5eafb/.github/ISSUE_TEMPLATE/Feature.md)
- I will submit a separate PR #??? shortly, which will improve `selfupdate` command and help:
  - The success message will say `Successfully updated rclone from OLDVER to NEWVER`
  - Synopsis will include instructions on reverting the update
  - Synopsis will include details about beta updates (e.g. "Some issues/features may only be fixed/added in the latest beta
with a point release pending")

**Review by darthShadow — APPROVED** [review]
LGTM.

Thanks a lot for doing this.

**Comment by ivandeex** [comment]
Commits squashed

**Comment by ivandeex** [comment]
@ncw

This PR updates the github ticket templates and its changes go live right after merge.
As only you decide when the next release happens, let me ask you.
Please merge this simultaneously with the next release.
@darthShadow and myself are happy with the patch.

Thank you.

**Comment by ncw** [comment]
Thanks! I've made a v1.55 release ticket which I've put a pointer to this in. #5162 - would appreciate feedback there!

**Pull request merged**
